flake.nix

```diff
@@ -162,60 +162,80 @@
-      postFixup = ''
-        wrapQtApp "$out/bin/artiq_dashboard"
-        wrapQtApp "$out/bin/artiq_browser"
-        wrapQtApp "$out/bin/artiq_session"
-      '';
+        preFixup = pkgs.lib.optionalString (!withMinimalDeps) ''
+          # Ensure that wrapProgram uses makeShellWrapper rather than makeBinaryWrapper
+          # brought in by wrapQtAppsHook. Only makeShellWrapper supports --run.
+          wrapProgram() { wrapProgramShell "$@"; }
+        '';
+        ## Modifies PATH to pass the wrapped python environment (i.e. python3.withPackages(...) to subprocesses.
+        ## Allows subprocesses using python to find all packages you have installed
+        makeWrapperArgs = pkgs.lib.optionals (!withMinimalDeps) [
+          ''--run 'if [ ! -z "$NIX_PYTHONPREFIX" ]; then export PATH=$NIX_PYTHONPREFIX/bin:$PATH;fi' ''
+          "--set FONTCONFIG_FILE ${pkgs.fontconfig.out}/etc/fonts/fonts.conf"
+        ];
 
-      preFixup = ''
-        # Ensure that wrapProgram uses makeShellWrapper rather than makeBinaryWrapper
-        # brought in by wrapQtAppsHook. Only makeShellWrapper supports --run.
-        wrapProgram() { wrapProgramShell "$@"; }
-      '';
-      ## Modifies PATH to pass the wrapped python environment (i.e. python3.withPackages(...) to subprocesses.
-      ## Allows subprocesses using python to find all packages you have installed
-      makeWrapperArgs = [
-        ''--run 'if [ ! -z "$NIX_PYTHONPREFIX" ]; then export PATH=$NIX_PYTHONPREFIX/bin:$PATH;fi' ''
-        "--set FONTCONFIG_FILE ${pkgs.fontconfig.out}/etc/fonts/fonts.conf"
-      ];
+        dontCheckRuntimeDeps = withMinimalDeps;
+        # FIXME: automatically propagate lld_20 llvm_20 dependencies
+        # cacert is required in the check stage only, as certificates are to be
+        # obtained from system elsewhere
+        nativeCheckInputs = pkgs.lib.optionals (!withMinimalDeps) [
+          pkgs.lld_20 pkgs.llvm_20 pkgs.lit pkgs.outputcheck pkgs.cacert libartiq-support
+        ];
+        checkPhase = pkgs.lib.optionalString (!withMinimalDeps) ''
+          python -m unittest discover -v artiq.test
 
-      # FIXME: automatically propagate lld_20 llvm_20 dependencies
-      # cacert is required in the check stage only, as certificates are to be
-      # obtained from system elsewhere
-      nativeCheckInputs = with pkgs; [lld_20 llvm_20 lit outputcheck cacert] ++ [libartiq-support];
-      checkPhase = ''
-        python -m unittest discover -v artiq.test
+          TESTDIR=`mktemp -d`
+          cp --no-preserve=mode,ownership -R $src/artiq/test/lit $TESTDIR
+          LIBARTIQ_SUPPORT=`libartiq-support` lit -v $TESTDIR/lit
+        '';
+      };
 
-        TESTDIR=`mktemp -d`
-        cp --no-preserve=mode,ownership -R $src/artiq/test/lit $TESTDIR
-        LIBARTIQ_SUPPORT=`libartiq-support` lit -v $TESTDIR/lit
-      '';
+    # Stripped down version of ARTIQ for gateware/firmware builds
+    artiq-minimal = artiq-upstream {
```

Comment:
``artiq-minimal`` imo can be misunderstood as minimum ARTIQ setup for actual usage, how about ``artiq-build``?